src/layers/point-layer/point-layer.js

```diff
@@ -138,9 +140,34 @@ export default class PointLayer extends Layer {
     return props;
   }
 
-  // TODO: fix complexity
-  /* eslint-disable complexity */
-  formatLayerData(allData, filteredIndex, oldLayerData, opt = {}) {
+  getChangedTriggers(dataUpdateTriggers) {
+    const triggerChanged = diffUpdateTriggers(dataUpdateTriggers, this._oldDataUpdateTriggers);
+    this._oldDataUpdateTriggers = dataUpdateTriggers;
+
+    return triggerChanged;
+  }
+
+  calculateDataAttribute(allData, filteredIndex, getPosition) {
+    const data = [];
+    for (let i = 0; i < filteredIndex.length; i++) {
```

Comment:
you can replace this loop with a reduce statement
```
const data = filteredIndex.reduce((acc, index) => {
...
}, []);
```

Reply:
This is a big loop that goes through the entire data array. `for loop` is more performative than `reduce`. I have decided to always use `for loop` when the data array needs to be iterated